Source/Meadow.Foundation.Peripherals/Sensors.Temperature.Max31865/Driver/Max31865.cs

```diff
@@ -0,0 +1,425 @@
+﻿using Meadow.Hardware;
+using Meadow.Peripherals.Sensors;
+using Meadow.Units;
+using System;
+using System.Threading.Tasks;
+
+namespace Meadow.Foundation.Sensors.Temperature;
+
+/// <summary>
+/// Driver class for the MAX31865 resistance-to-digital converter.
+/// Provides temperature readings from Resistance Temperature Detectors such as the PT100 and PT1000.
+/// </summary>
+/// <remarks>
+/// This class implements the ITemperatureSensor interface for temperature readings
+/// and ISpiPeripheral interface for SPI communication. It follows the IDisposable pattern
+/// for proper resource management.
+/// </remarks>
+public class Max31865 : SamplingSensorBase<Temperature>, ITemperatureSensor, ISpiPeripheral, IDisposable
```

Comment:
I have added Sampling support into this sensor, is this ok or going too far for a v1?

Reply:
Sampling is fine.  It largely depends on how the consumer uses it, though my feeling is that sampling is likely outside the purpose of a driver unless the hardware itself provides it.  We're going to improve the SensorService to provide sampling at some point, but for now this is fine.

Reply:
I'm happy to remove it as I get your point about the driver level

Reply:
Yeah, ITemperatureSensor shouldn't (And I think in the latest doesn't) implement ISamplingSensor.

Reply:
I have moved away from sampling now